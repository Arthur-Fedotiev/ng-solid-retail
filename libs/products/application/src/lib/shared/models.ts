import { Categories, Colors, Retailers, Tiers } from './constants';

export type CategoriesMap = typeof Categories;
export type Category = typeof Categories[keyof typeof Categories];

export type RetailersMap = typeof Retailers;
export type TiersMap = typeof Tiers;
export type ColorsMap = typeof Colors;
