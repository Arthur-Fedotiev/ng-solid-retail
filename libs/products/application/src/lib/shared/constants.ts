import {
  ProductCategory,
  ProductRetailer,
  ProductTier,
  SHOES_AND_CLOTHING_COLORS,
} from '@sr/products/entities';

export const Categories = ProductCategory;
export const Retailers = ProductRetailer;
export const Tiers = ProductTier;
export const Colors = {
  [Categories.Shoes]: SHOES_AND_CLOTHING_COLORS,
  [Categories.Clothing]: SHOES_AND_CLOTHING_COLORS,
} as const;
