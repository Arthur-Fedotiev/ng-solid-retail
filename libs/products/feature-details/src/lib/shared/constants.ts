import { Tiers } from '@sr/products/application';

export const PRODUCT_TIERS = Tiers;
