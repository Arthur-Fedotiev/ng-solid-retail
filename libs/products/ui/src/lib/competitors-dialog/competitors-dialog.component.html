<h1 mat-dialog-title>Competitors for {{ data.category.name }}</h1>
<div mat-dialog-content>
  <ul *ngFor="let competitor of data.retailers$ | async">
    <li>
      <span>{{ competitor.name }}</span>
    </li>
  </ul>
</div>
