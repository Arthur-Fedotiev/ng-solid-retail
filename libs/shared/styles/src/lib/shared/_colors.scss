$colors: (
  'M900': #1b1c1d,
  'M800': #4e5052,
  'M700': #969798,
  'M600': #f1f3f4,
  'M500': #727779,
  'M300': #d9dadc,
  'M100': #f1f3f4,
  'M0': #ffffff,
  'S50': #fafdfd,
  'S100': #f0f8f8,
  'B900': #4b79f7,
  'R50': #ffebee,
  'R100': #ffcacf,
  'R500': #fa3126,
  'R600': #eb2327,
  'R700': #d91421,
);

@function app-color($color) {
  @return map-get($colors, $color);
}
