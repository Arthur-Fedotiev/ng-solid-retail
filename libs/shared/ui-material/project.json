{
  "name": "shared-ui-material",
  "projectType": "library",
  "sourceRoot": "libs/shared/ui-material/src",
  "prefix": "sr",
  "targets": {
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/libs/shared/ui-material"],
      "options": {
        "jestConfig": "libs/shared/ui-material/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": [
          "libs/shared/ui-material/**/*.ts",
          "libs/shared/ui-material/**/*.html"
        ]
      }
    }
  },
  "tags": ["scope:shared", "type:ui"]
}
