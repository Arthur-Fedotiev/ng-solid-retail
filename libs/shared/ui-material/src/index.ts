export * from './lib/shared-ui-material.module';
