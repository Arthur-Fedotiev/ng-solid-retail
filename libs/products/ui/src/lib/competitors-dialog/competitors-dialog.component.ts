import { ChangeDetectionStrategy, Component, Inject } from '@angular/core';
import { MAT_LEGACY_DIALOG_DATA as MAT_DIALOG_DATA } from '@angular/material/legacy-dialog';
import { TRACK_BY_ID_OR_IDX, TrackByIdOrIdx } from '@sr/shared/util';
import { Observable } from 'rxjs';

interface CompetitorsDialogData {
  category: { name: string };
  retailers$: Observable<{ name: string; id: string }[]>;
}

@Component({
  selector: 'sr-competitors-dialog',
  templateUrl: './competitors-dialog.component.html',
  styleUrls: ['./competitors-dialog.component.scss'],
  changeDetection: ChangeDetectionStrategy.OnPush,
})
export class CompetitorsDialogComponent {
  constructor(
    @Inject(MAT_DIALOG_DATA) public data: CompetitorsDialogData,
    @Inject(TRACK_BY_ID_OR_IDX) public readonly trackById: TrackByIdOrIdx
  ) {}
}
