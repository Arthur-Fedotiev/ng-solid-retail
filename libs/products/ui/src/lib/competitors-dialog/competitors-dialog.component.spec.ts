import { ComponentFixture, TestBed } from '@angular/core/testing';
import { MAT_DIALOG_DATA } from '@angular/material/dialog';
import { SharedUiMaterialModule } from '@omnia/shared/ui-material';
import { of } from 'rxjs';

import { CompetitorsDialogComponent } from './competitors-dialog.component';

describe('CompetitorsDialogComponent', () => {
  let component: CompetitorsDialogComponent;
  let fixture: ComponentFixture<CompetitorsDialogComponent>;

  beforeEach(async () => {
    await TestBed.configureTestingModule({
      imports: [SharedUiMaterialModule],
      providers: [
        {
          provide: MAT_DIALOG_DATA,
          useValue: {
            category: { id: '1', name: 'Category 1' },
            competitors$: of([{ id: '1', name: 'Competitor 1' }]),
          },
        },
      ],
      declarations: [CompetitorsDialogComponent],
    }).compileComponents();

    fixture = TestBed.createComponent(CompetitorsDialogComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should render the competitors', () => {
    expect(fixture).toMatchSnapshot();
  });
});
