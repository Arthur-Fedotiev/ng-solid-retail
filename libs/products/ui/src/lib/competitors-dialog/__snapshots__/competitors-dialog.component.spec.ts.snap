// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CompetitorsDialogComponent should render the competitors 1`] = `
<sr-competitors-dialog
  __ngContext__={[Function Number]}
  data={[Function Object]}
  trackById={[Function Function]}
>
  <h1
    class="mat-dialog-title"
    mat-dialog-title=""
  >
    Competitors for Category 1
  </h1><div
    class="mat-dialog-content"
    mat-dialog-content=""
  >
    <mat-progress-spinner
      aria-valuemax="100"
      aria-valuemin="0"
      aria-valuenow="75"
      class="mat-progress-spinner mat-spinner mat-primary mat-progress-spinner-indeterminate-animation"
      color="primary"
      mode="determinate"
      role="progressbar"
      style="width: 100px; height: 100px;"
      tabindex="-1"
    >
      <svg
        aria-hidden="true"
        focusable="false"
        preserveAspectRatio="xMidYMid meet"
        style="width: 100px; height: 100px;"
        viewBox="0 0 100 100"
      >
        <circle
          cx="50%"
          cy="50%"
          r="45"
          style="stroke-dashoffset: 70.68583470577035px; stroke-dasharray: 282.7433388230814px; stroke-width: 10%; transform-origin: 50% 50%;"
        />
      </svg>
    </mat-progress-spinner>
  </div>
</sr-competitors-dialog>
`;
